torchvision/transforms/transforms.py

```diff
@@ -1375,3 +1375,47 @@ def __call__(self, img):
             x, y, h, w, v = self.get_params(img, scale=self.scale, ratio=self.ratio, value=self.value)
             return F.erase(img, x, y, h, w, v, self.inplace)
         return img
+
+
+class AddGaussianNoise(object):
+    """Add gaussian noise to a given tensor with probability p. More specifically,
+    for each value in the tensor, add a random number sampled from a Gaussian
+    with the given mean and std.
+
+    Useful to apply to images during the training phase to reduce overfitting.
+
+    Args:
+        mean (float): mean of the Gaussian distribution from which to sample.
+        std (float): std of the Gaussian distribution from which to sample.
+        p (float): probability that noise is added.
+
+    Returns:
+        Tensor with Gaussian noise added.
+    """
+
+    def __init__(self, mean=0., std=1., p=1.):
```

Comment:
I know that the parameter order is all over the place across these classes, but would it make sense for p to be the first one?
Also, maybe a default of 0.5 would be more consistent with the other transforms.